autoscale-groups-core/src/main/java/fish/payara/extensions/autoscale/groups/core/admin/ScaleCommandHelper.java

```diff
@@ -0,0 +1,327 @@
+        }
+
+        LOGGER.info(String.format(
+                "Executing %s on %d instances using a thread pool of size %d: %s", commandName, nInstances,
+                threadPoolSize, targetServers.stream().map(Server::getName).collect(Collectors.joining(", "))));
+
+        progressStatus.setTotalStepCount(nInstances);
+        progressStatus.progress(Strings.get("cluster.command.executing", commandName, nInstances));
+
+        // Loop through instance names, construct the command for each instance name, and hand it off to the threadpool.
+        for (Server server : targetServers) {
+            String serverName = server.getName();
+            waitingForServerNames.add(serverName);
+
+            ParameterMap instanceParameterMap = new ParameterMap(parameterMap);
+            // Set the instance name as the operand for the commnd
+            instanceParameterMap.set("DEFAULT", serverName);
+
+            ActionReport instanceReport = actionReport.addSubActionsReport();
+            instanceReport.setActionExitCode(SUCCESS);
+            CommandRunner.CommandInvocation invocation = commandRunner.getCommandInvocation(commandName, instanceReport,
+                    subject);
+            invocation.parameters(instanceParameterMap);
+
+            String msg = commandName + " " + serverName;
+            LOGGER.info(msg);
+
+            // Wrap the command invocation in a runnable and hand it off
+            // to the thread pool
+            CommandRunnable cmdRunnable = new CommandRunnable(invocation, instanceReport, responseQueue);
+            cmdRunnable.setName(serverName);
+            threadPool.execute(cmdRunnable);
+        }
+
+        if (LOGGER.isLoggable(FINE)) {
+            LOGGER.fine(String.format("%s commands queued, waiting for responses", commandName));
+        }
+
+        // Make sure we don't wait longer than the admin read timeout. Set our limit to be 3 seconds less.
+        long adminTimeout = RemoteRestAdminCommand.getReadTimeout() - 3000;
+        if (adminTimeout <= 0) {
+            // This should never be the case
+            adminTimeout = 57 * 1000;
+        }
+
+        if (LOGGER.isLoggable(FINE)) {
+            LOGGER.fine(String.format("Initial cluster command timeout: %d ms", adminTimeout));
+        }
+
+        // Now go get results from the response queue.
+        for (int n = 0; n < nInstances; n++) {
+            long timeLeft = adminTimeout - (System.currentTimeMillis() - startTime);
+            if (timeLeft < 0) {
+                timeLeft = 0;
+            }
+            CommandRunnable cmdRunnable = null;
+            try {
+                cmdRunnable = responseQueue.poll(timeLeft, MILLISECONDS);
+            } catch (InterruptedException e) {
+                // This thread has been interrupted. Abort
+                threadPool.shutdownNow();
+                String msg = Strings.get("cluster.command.interrupted", targetNames, n, nInstances, commandName);
```

Comment:
This may reference the wrong message string, see my previous comment

Reply:
`Command for target {0} was interrupted after getting responses from {1} of {2} instances for command {3}.`